connectors/sources/zoom.py

```diff
@@ -0,0 +1,539 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Zoom source module responsible to fetch documents from Zoom."""
+import asyncio
+import os
+from contextlib import asynccontextmanager
+from datetime import datetime, timedelta
+from functools import cached_property, partial
+
+import aiofiles
+import aiohttp
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
+from aiohttp.client_exceptions import ClientResponseError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CacheWithTimeout,
+    CancellableSleeps,
+    RetryStrategy,
+    convert_to_b64,
+    get_base64_value,
+    iso_utc,
+    retryable,
+)
+
+RETRIES = 3
+RETRY_INTERVAL = 2
+CHAT_PAGE_SIZE = 50
+MEETING_PAGE_SIZE = 300
+FILE_SIZE_LIMIT = 10485760
+
+if "OVERRIDE_URL" in os.environ:
+    override_url = os.environ["OVERRIDE_URL"]
+    BASE_URL = override_url
+    BASE_AUTH_URL = override_url
+else:
+    BASE_URL = "https://api.zoom.us/v2"
+    BASE_AUTH_URL = "https://zoom.us"
+
+AUTH = (
+    "{base_auth_url}/oauth/token?grant_type=account_credentials&account_id={account_id}"
+)
+APIS = {
+    "USERS": "{base_url}/users?page_size={page_size}",
+    "MEETINGS": "{base_url}/users/{user_id}/meetings?page_size={page_size}&type={meeting_type}",
+    "PAST_MEETING": "{base_url}/past_meetings/{meeting_id}",
+    "PAST_MEETING_PARTICIPANT": "{base_url}/past_meetings/{meeting_id}/participants?page_size={page_size}",
+    "RECORDING": "{base_url}/users/{user_id}/recordings?page_size={page_size}&from={date_from}&to={date_to}",
+    "CHANNEL": "{base_url}/chat/users/{user_id}/channels?page_size={page_size}",
+    "CHAT": "{base_url}/chat/users/{user_id}/messages?page_size={page_size}&search_key=%20&search_type={chat_type}&from={date_from}&to={date_to}",
+}
+
+
+def format_recording_date(date):
+    return date.strftime("%Y-%m-%d")
+
+
+def format_chat_date(date):
+    return date.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+class TokenError(Exception):
+    pass
+
+
+class NotFound(Exception):
+    pass
```

Comment:
Nit: I think this could be more clearly named. I have to search through the code to understand what could raise this error.
Can you add a prefix, e.g. `ZoomResourceNotFound` (so we can understand it's an API error and not a connector error), or a comment explaining what could cause this error?